asf_core_data/notebooks/get_address_matching_sample.py

```diff
@@ -0,0 +1,318 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.15.2
+#   kernelspec:
+#     display_name: asf_core_data_kernel
+#     language: python
+#     name: asf_core_data_kernel
+# ---
+
+# %%
+import pandas as pd
+import os
+from datetime import datetime
+
+# %%
+from asf_core_data.pipeline.mcs.process import mcs_epc_joining as joining
+
+# %% [markdown]
+# ## Set local_core_data_dir where EPC data is stored
+
+# %%
+## Set your local ASF core data dir
+local_core_data_dir = ""
+
+## Output data will be saved here
+outputs_data_dir = os.path.abspath("../../outputs/data")
+
+# %% [markdown]
+# ## Read in data
+#
+# EPC data: 2023 Q2 complete, GB dedupl and preprocessed\
+# MCS data: mcs_installations_231009
+
+# %%
+epc_data = pd.read_csv(
+    os.path.join(
+        local_core_data_dir,
+        "outputs/EPC/preprocessed_data/2023_Q2_complete/EPC_GB_preprocessed_and_deduplicated.csv",
+    )
+)
+mcs_data = pd.read_csv("s3://asf-core-data/outputs/MCS/mcs_installations_231009.csv")
+
+# %%
+print(epc_data.shape)
+print(mcs_data.shape)
+
+# %% [markdown]
+# ## Prepare data for address matching
+
+# %%
+epc_df = joining.prepare_epcs(epc_data)
+
+# %%
+mcs_df = joining.prepare_hps(mcs_data)
```

Comment:
Yes, `prepare_hps` concatenates 'address_1` and `address_2` from mcs address fields and then standardises them with by removing punctuation and converting to lower case. `prepare_epcs` only standardises `ADDRESS1` field without concatenating any other address lines. I've now updated the `prepare_epcs` function to use `ADDRESS2` and will review the effect. I think it will have some impact but I think a lot of the initial errors discovered will still be present.